src/Application/Actions/Hooks/StripePaymentsUpdate.php

```diff
@@ -315,6 +318,33 @@ private function handleChargeRefunded(Event $event, Response $response): Respons
         return $this->respondWithData($response, $charge);
     }
 
+    private function handlePaymentIntentCancelled(Request $request, Event $event, Response $response): Response
+    {
+        /** @var PaymentIntent $paymentIntent */
+        $paymentIntent = $event->data->object;
+        $donation = $this->donationRepository->findOneBy(['transactionId' => $paymentIntent->id]);
+
+        if ($donation === null) {
+            return $this->respond($response, new ActionPayload(404));
+        }
+
+        if ($donation->getDonationStatus() !== DonationStatus::Cancelled) {
+            $this->logger->info(sprintf(
+                'Received Stripe cancellation request, Donation ID %s, payment intent ID %s',
+                $donation->getId() ?? throw new \Exception("Missing ID on donation"),
```

Comment:
Including donation UUID in all logs – instead of or in addition to numeric ID – makes it much easier to search all relevant activity in CloudWatch Logs Insights

Reply:
Thanks, done.